tensorzero-core/src/cache.rs

```diff
@@ -216,34 +218,40 @@ pub struct CacheData<T: CacheOutput> {
 /// (e.g. `infer_stream` will never try to deserialize a `NonStreamingCacheData`)
 pub trait CacheOutput {
     /// If this return `false`, then we'll log a warning and skip writing this entry to the cache
-    fn should_write_to_cache(&self) -> bool;
+    fn should_write_to_cache(
+        &self,
+        cache_validation_info: CacheValidationInfo,
+    ) -> impl Future<Output = bool> + Send;
 }
 
 impl CacheOutput for StreamingCacheData {
-    fn should_write_to_cache(&self) -> bool {
+    async fn should_write_to_cache(&self, _cache_validation_info: CacheValidationInfo) -> bool {
         true
```

Comment:
~~couldn't we run `collect_chunks` here and then validate?~~

ok just kidding, that would be a large refactor; let's add a comment here about why we're doing it this way and open a narrower follow-up issue that we may handle one day.